### PR TITLE
Week 3 assessment ato

### PR DESCRIPTION
**Pull request by HackCoder77**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,47 +16,116 @@
...
+        expect(cumSum(numbersToAdd3)).toEqual([])
+        
+    })
+})
 
 // b) Create the function that makes the test pass.
+//blank arrays return blank arrays
+//number 1 + number 2 = number i
+
+let cumSum = array => array.map((sum => value => sum += value)(0));
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:124]: Alright that makes a lot more sense when I don't have 5 ways I am trying to solve it. Still when we (anyone: junior mid or senior) are bringing in dense syntax we should take a moment to leave a note of how this syntax is work and why it is working. Line 122 does make sense in this regard but expand on it a bit more and talk about why its a better methodology.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,47 +16,116 @@
...
 
+// make array length be the number of input. I did not need to do this but i originally thought i would. i just set the loop to num.
+//number 1 and number 2 must = number 3
+//take 3rd number and make it equal to itself minus 2 and minus 1 to get number 3.
+//loop thru and make sure to go length of num
+
+function fibonacci(num) {
+    let array = [];
+    array[0] = 1;
+    array[1] = 1;
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:44]: yeah easy refactor into 
```javascript
    let array = [1,1]
```

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,47 +16,116 @@
...
+        if(typeof array[i] == 'number' && array[i] % 2 === 1){
+            final.push(array[i])
+         } 
+         if(array[i] < 0){
+             Math.abs(array[i])
+             final.push(array[i])
+         }
+        }
+    return final.sort((a, b) => a - b);
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:87]: Great work!

File: interview-questions.md [hunk 4]
```diff
@@ -9,59 +9,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a testing application for your code.  Files aren't modified but they are ran for testing. So the app.js file is ran.
 
-  Researched answer:
+  Researched answer: package managers that help to manage a project's dependencies. node modules, yarn lock, package.json.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:47]: Yea, Yarn does a lot of things. Companies are pretty particular about choosing the package managers they use. Hence why Facebook created Yarn in the first place!

**Review by WalkerTexas-coder — APPROVED** [review]
Ato, great work. Your code logic is on point and you already have a lot of skill to showcase. I reccommend on doing what you can to explain the code logic to others. The difference between a junior, mid and senior level developer is not just the ability to code but the ability to teach and communicate with others. If you want to level up your skill that's what you should be focusing on. That'll be worth significant value and monetary compensation. Since I know you can do it, I'll be looking for well written and thoughtful pseudo code next week! this PR is approved!

File: code-challenges.test.js (hunk 1 above)
Comment (HackCoder77) [reply code-challenges.test.js:124]: Thanks for the feedback Austin. Especially if we talking about getting me some more money!! haha appreciate it

File: code-challenges.test.js (hunk 2 above)
Comment (HackCoder77) [reply code-challenges.test.js:44]: Got it. its funny because i see why it works but it doesn't feel natural to me to write it like that

**Pull request merged**